R/utils.R

```diff
@@ -132,6 +132,19 @@ hasPrefix <- function(x, prefix) {
   substring(x, 1, nchar(prefix)) == prefix
 }
 
+# Lightweight equivalent of withr::defer()
+defer <- function(expr, env = caller_env(), after = FALSE) {
+  thunk <- as.call(list(function() expr))
+  do.call(on.exit, list(thunk, TRUE, after), envir = env)
```

Comment:
Nit: The `after` argument requires R (>= 3.5.0); can we take a dependency on newer R here?

Reply:
We already have an implicit dependency on 3.5 via rlang, but I've made the explicit in the `DESCRIPTION`.